File: AdminPanal/AdminPanal.Web/Modules/ServerTypes/AylaStoreModule/CategoryRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface CategoryRow {
    Id?: number;
    Name?: string;
    Description?: string;
    Photo?: string;
}

export abstract class CategoryRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'AylaStoreModule.Category';
    static readonly lookupKey = 'AylaStoreModule.Category';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<CategoryRow>('AylaStoreModule.Category') }
    static async getLookupAsync() { return getLookupAsync<CategoryRow>('AylaStoreModule.Category') }

    static readonly deletePermission = 'Category';
    static readonly insertPermission = 'Category';
    static readonly readPermission = 'Category';
    static readonly updatePermission = 'Category';

    static readonly Fields = fieldsProxy<CategoryRow>();
}

File: AdminPanal/AdminPanal.Web/Modules/AylaStoreModule/Category/CategoryRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace AdminPanal.AylaStoreModule;

[ConnectionKey("Ayla_Store_MVC_Database"), Module("AylaStoreModule"), TableName("Category")]
[DisplayName("Category"), InstanceName("Category")]
[ReadPermission("Category")]
[ModifyPermission("Category")]

[LookupScript]
public sealed class CategoryRow : Row<CategoryRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id
    {
        get => fields.Id[this];
        set => fields.Id[this] = value;
    }

    [DisplayName("Name"), Size(50), QuickSearch, NameProperty]
    public string Name
    {
        get => fields.Name[this];
        set => fields.Name[this] = value;
    }

    [DisplayName("Description")]
    public string Description
    {
        get => fields.Description[this];
        set => fields.Description[this] = value;
    }

    [DisplayName("Photo")]
    public string Photo
    {
        get => fields.Photo[this];
        set => fields.Photo[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField Name;
        public StringField Description;
        public StringField Photo;

    }
}